src/yoroi-wallets/cardano/transactionManager.ts

```diff
@@ -0,0 +1,62 @@
+import {Storage} from '../storage'
+import {BackendConfig} from '../types'
+import {TransactionCache} from './shelley'
+
+export const makeTransactionManager = async (storage: Storage) => {
+  const transactionCache = await TransactionCache.create(storage.join('txs/'))
+  const memosManager = await makeMemosManager(storage.join('memos/'))
+
+  return {
+    // transactionCache api
+    get transactions() {
+      const {memos} = memosManager
+      return Object.keys(transactionCache.transactions).reduce(
+        (result, current) => ({
+          ...result,
+          [current]: {
+            ...transactionCache.transactions[current],
+            memo: memos[current],
+          },
+        }),
+        {},
+      )
+    },
+    get perRewardAddressCertificates() {
+      return transactionCache.perRewardAddressCertificates
+    },
+    get perAddressTxs() {
+      return transactionCache.perAddressTxs
+    },
+    get confirmationCounts() {
+      return transactionCache.confirmationCounts
+    },
+    clear: () => transactionCache.clear(),
+    resetState: () => transactionCache.resetState(),
+    subscribe: (handler) => transactionCache.subscribe(handler),
+    doSync: async (addressesByChunks: Array<Array<string>>, backendConfig: BackendConfig) =>
```

Comment:
can we move the `backendConfig` to the initialization?

Reply:
I have checked it I have noticed that the transaction manager instance is created and passed as an argument to shelley wallet instance creation. 

Do you think should the transaction manager instance be created after the shelley wallet instance is created?

Reply:
I think this should be part of `commonCreate()` as UtxoManager is.

Reply:
backend config is synchronously available there, so we don't need to provide it every time.

Reply:
Correct me if I am wrong! but `commonCreate` is used in static methods and `getBackendConfig` is a method from `ShelleyWallet` so it is not possible get the backend config unless its obtained independently of `getBackendConfig`

Reply:
there is a version, `getNetworkConfigById`, I believe they return the same thing based on networkId

Reply:
you are right ! 👍🏻